diffpy/snmf/factorizers.py

```diff
@@ -0,0 +1,29 @@
+import numpy as np
+import scipy.optimize
+
+
+def lsqnonneg(coefficient, target):
+    """
+    Solve ``argmin_x || Ax - b ||_2`` for ``x>=0``.
+
+    Parameters
+    ----------
+    coefficient: 2-dimensional ndarray or list
+        The coefficient matrix of the least squares problem
+    target: 1-dimensional ndarray or list
+        The target vector for the least squares problem
```

Comment:
also, `target` is a bit confusing to me here.   What does this mean?  What should I be giving this function?  Also, what is special about it being least-squares, That could be mentioned, but least-squares is not  a type of problem but a regression method.  Maybe this is easier to sort out on a live call.